src/scicookie/{{cookiecutter.project_slug}}/build-system/hatch-pyproject.toml

```diff
@@ -0,0 +1,128 @@
+[build-system]
+requires = ["hatchling>=1.17.1"]
+build-backend = "hatchling.build"
+
+[project]
+name = "{{ cookiecutter.project_slug }}"
+authors = [
+  { name = "{{ cookiecutter.author_full_name }}", email = "{{ cookiecutter.author_email }}" },
+]
+description = "{{ cookiecutter.project_short_description }}"
+readme = "README.md"
+classifiers = [
+  "Development Status :: 1 - Planning",
+  "Intended Audience :: Science/Research",
+  "Intended Audience :: Developers",
+{%- if cookiecutter.project_license == "MIT" %}
+  "License :: OSI Approved :: MIT License",
+{%- elif cookiecutter.project_license == "BSD 3 Clause" %}
+  "License :: OSI Approved :: BSD License",
+{%- elif cookiecutter.project_license == "Apache Software License 2.0" %}
+  "License :: OSI Approved :: Apache Software License",
+{%- elif cookiecutter.project_license == "GNU General Public License v3" %}
+  "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
+{%- endif %}
+  "Operating System :: OS Independent",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Topic :: Scientific/Engineering",
+  "Typing :: Typed",
+]
+version = "{{ cookiecutter.project_version }}"
```

Comment:
To be honest, I am not a big fan of static versioning and neither is this repository. It looks like all the Pythonic backends (except poetry, probably because it does not support it by default) in this repository use dynamic versioning, which makes it natural to make Hatch use dynamic versioning too.

The final call would be of the maintainers.

Reply:
In parts I agree about the static vs dynamic versioning 
but using semantic-release,  although you will end up with the versions static there, but you can use semantic-release-replace to update the versions automatically. and semantic release is a full stack for releasing ... different than the scm approaches ... maybe it would be possible to use semantic release + any scm approach .. but I never tested it to be honest. but also IIRC poetry doesn't have a good alternative for scm).

with all this said, what I meant is: for now it uses semantic release by default ...
so what we would need to have is (not for this PR):
- an option for static or dynamic versioning
- and an option for releasing tool, and can start with just semantic-release 

for this PR I think that static versioning is ok, so we can keep consistency with all the other build systems and we can open an issue about that and start some discussions to check what we need for supporting both static and dynamic options.

PS: the idea is to be open to different approach as much as possible .. and we can have different "profiles" that could be more opinionated